### PR TITLE
Sanitize filenames on upload endpoints

### PR DESCRIPTION
**Pull request by LVMBDV**
Part of a fix for SERVER-1713 SERVER-1728.

### CODE REVIEW

**Comment by ar2rsawseen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5176435&bid=16990658)

File: api/utils/common.js [hunk 1]
```diff
@@ -2283,4 +2283,22 @@ common.checkDatabaseConfigMatch = (apiConfig, frontendConfig) => {
...
+        return "";
+    }
+
+    return filename
+        .replace(/[\x00-\x1f\x80-\x9f]+/g, replacement)
+        .replace(/[\/\?<>\\:\*\|"]/g, replacement)
+        .replace(/^\.{1,2}$/, replacement)
+        .replace(/^\.+/, replacement);
+};
+
```
Comment (ar2rsawseen) [inline api/utils/common.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5176435)

File: api/utils/common.js [hunk 2]
```diff
@@ -2283,4 +2283,22 @@ common.checkDatabaseConfigMatch = (apiConfig, frontendConfig) => {
...
+ * @param {string} filename -
+ * @param {string} replacement -
+ * @returns {string} sanitizedFilename -
+ */
+common.sanitizeFilename = (filename, replacement = "") => {
+    if (typeof filename !== "string") {
+        return "";
+    }
+
+    return filename
```
Comment (ar2rsawseen) [inline api/utils/common.js]: Codacy Issue found: [Unexpected control character(s) in regular expression: \x00, \x1f.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5176435)

**Comment by ar2rsawseen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5176435&bid=16990658)

File: api/utils/common.js [hunk 3]
```diff
@@ -2283,4 +2283,22 @@ common.checkDatabaseConfigMatch = (apiConfig, frontendConfig) => {
...
+ * @param {string} replacement - string to replace characters to be removed
+ * @returns {string} sanitizedFilename - sanitized filename
+ */
+common.sanitizeFilename = (filename, replacement = "") => {
+    if (typeof filename !== "string") {
+        return "";
+    }
+
+    return filename
+        .replace(/[\x00-\x1f\x80-\x9f]+/g, replacement)
```
Comment (ar2rsawseen) [inline api/utils/common.js]: Codacy Issue found: [Unexpected control character(s) in regular expression: \x00, \x1f.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5176435)

Comment (LVMBDV) [reply api/utils/common.js]: yes, that is the point :smile:

**Comment by ar2rsawseen** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5176435&bid=16990658)

File: api/utils/common.js (hunk 3 above)
Comment (ar2rsawseen) [inline api/utils/common.js]: Codacy Issue found: [Unexpected control character(s) in regular expression: \x00, \x1f.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5176435)

**Pull request merged**